storm-server/src/main/java/org/apache/storm/scheduler/resource/strategies/scheduling/ConstraintSolverStrategy.java

```diff
@@ -248,10 +247,22 @@ public SchedulingResult schedule(Cluster cluster, TopologyDetails td) {
         nodes = RAS_Nodes.getAllNodesFrom(cluster);
         Map<WorkerSlot, Set<String>> workerCompAssignment = new HashMap<>();
         Map<RAS_Node, Set<String>> nodeCompAssignment = new HashMap<>();
-        //set max number of states to search
-        final int maxStateSearch = Math.min(MAX_STATE_SEARCH,
-                                            ObjectReader.getInt(td.getConf().get(Config.TOPOLOGY_RAS_CONSTRAINT_MAX_STATE_SEARCH),
-                                                                DEFAULT_STATE_SEARCH));
```

Comment:
We still need defaults.

Reply:
The default has been moved to defaults.yaml as a DaemonConfig

Reply:
The DEFAULT_STATE_SEARCH was added as Config.TOPOLOGY_RAS_CONSTRAINT_MAX_STATE_SEARCH; The larger MAX_STATE_SEARCH was added as DaemonConfig.RESOURCE_AWARE_SCHEDULER_MAX_STATE_SEARCH.

Reply:
cool